drivers/sensor/bmi160/bmi160_v2.c

```diff
@@ -0,0 +1,151 @@
+/* Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/math/util.h>
+#include <zephyr/sys_clock.h>
+
+#include "bmi160.h"
+
+static int bmi160_read_data(const struct device *sensor, struct sensor_data *out)
+{
+	struct bmi160_data *data = sensor->data;
+	int rc;
+	uint16_t *sample;
+
+	out->header.base_timestamp = k_uptime_get() * USEC_PER_MSEC;
```

Comment:
k_uptime_get and k_cycle_get won't get you nanosecond resolutions but the header and comments seem to suggest you want that